app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -932,7 +935,16 @@ public void onResume() {
             thumbnailEditIcon.setColorFilter(Utils.needsDarkForeground(headerColor) ? Color.WHITE : Color.BLACK);
         }
 
+        // Set the balance directly in case the locale (and thus the decimal separator) changed,
+        // since that may cause parseBalance() to return wrong values
+        BigDecimal balance = tempLoyaltyCard.balance == null ? new BigDecimal("0") : tempLoyaltyCard.balance;
+        tempLoyaltyCard = updateTempState(tempLoyaltyCard, LoyaltyCardField.balance, balance);
+        balanceField.setText(Utils.formatBalanceWithoutCurrencySymbol(tempLoyaltyCard.balance, tempLoyaltyCard.balanceType));
+        validBalance = true;
+        Log.d(TAG, "Setting balance (in case locale changed) to " + balance);
+
         onResuming = false;
+        onRestoring = false;
```

Comment:
I get what you were going for, but I know myself and if it says "Do X in case of Y" my brain tends to interpret it as an `if (Y) { X() }`. So I think using other phrasing would help me in the future. A suggestion:

```suggestion
        // Set the balance
        // We do this before finishing `onResuming` and `onRestoring` to prevent `formatBalanceCurrencyField` from triggering a reformat (which can cause issues when switching locale)
        BigDecimal balance = tempLoyaltyCard.balance == null ? new BigDecimal("0") : tempLoyaltyCard.balance;
        tempLoyaltyCard = updateTempState(tempLoyaltyCard, LoyaltyCardField.balance, balance);
        balanceField.setText(Utils.formatBalanceWithoutCurrencySymbol(tempLoyaltyCard.balance, tempLoyaltyCard.balanceType));
        validBalance = true;
        Log.d(TAG, "Setting balance to " + balance);

        onResuming = false;
        onRestoring = false;
```

Reply:
Do we maybe want to pull line 832 (`formatBalanceCurrencyField(tempLoyaltyCard.balanceType);` down here to put all balance-related operations together?

Reply:
I didn't think that was needed since the `formatBalanceCurrencyField` now no longer affects the balance field during this function. But I can move it if you prefer.

Reply:
I'll update the comment too.

Reply:
Yeah I know it's not necessary, but I find the code easier to follow is the balance-related operations are grouped, is all.

Reply:
If I move `formatBalanceCurrencyField(tempLoyaltyCard.balanceType)` down, the `startWithoutParametersViewBack` test fails. So I moved everything up.

Reply:
That makes more sense anyway. Should probably not have put it below the `initDone`, but it worked fine before I moved that one too.